src/server/game/Maps/Map.cpp

```diff
@@ -2478,12 +2485,20 @@ bool Map::GetObjectHitPos(uint32 phasemask, float x1, float y1, float z1, float
     return result;
 }
 
-float Map::GetHeight(uint32 phasemask, float x, float y, float z, bool vmap/*=true*/, float maxSearchDist /*= DEFAULT_HEIGHT_SEARCH*/) const
+float Map::GetHeight(uint32 phasemask, float x, float y, float z, bool vmap /*= true*/, float maxSearchDist /*= DEFAULT_HEIGHT_SEARCH*/) const
 {
-    float h1, h2;
-    h1 = GetHeight(x, y, z, vmap, maxSearchDist);
-    h2 = _dynamicTree.getHeight(x, y, z, maxSearchDist, phasemask);
-    return std::max<float>(h1, h2);
+    DynamicTreeCallback* dCallback = nullptr;
+    float vmapZ = GetHeight(x, y, z, vmap, maxSearchDist);
+    float goZ   = _dynamicTree.getHeight(x, y, z, maxSearchDist, phasemask, dCallback);
+    if (vmapZ > goZ && dCallback)
```

Comment:
`dCallback` is always null - what this changes?

Reply:
Stops the undermeshing.